apps/blog/src/app/form/form.component.ts

```diff
@@ -34,17 +36,19 @@ export class FormComponent {
   error = false;
 
   constructor(private http: HttpClient,
-              private formService: FormService
+              private postService: PostService,
+              private router: Router
   ) { }
 
   onSubmit() {
     const formValues: any = this.myform.getRawValue();
-    this.formService
+    this.postService
       .addPost(
         formValues
       )
-      .then(() => {
+      .then((result) => {
```

Comment:
nit: you can do
`then(({key}) => {`

 this.router.navigateByUrl(`post/${key}`);